presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -551,7 +552,34 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         if (needPostProjectionCoercion) {
             return explicitCoercionFields(subPlan, distinctGroupingColumns, analysis.getAggregates(node));
         }
-        return subPlan;
+
+        // 4. Project and re-write all grouping functions
+        return handleGroupingOperations(subPlan, node, groupIdSymbol);
+    }
+
+    private PlanBuilder handleGroupingOperations(PlanBuilder subPlan, QuerySpecification node, Optional<Symbol> groupIdSymbol)
+    {
+        if (analysis.getGroupingOperations(node).isEmpty()) {
+            return subPlan;
+        }
+
+        TranslationMap translations = subPlan.copyTranslations();
```

Comment:
I don't understand this suggestion. Doesn't this contradict your comment here https://github.com/prestodb/presto/pull/6704#discussion_r107416340? If not can you clarify what you mean?